### ISSUE TITLE
"Add word to CSpell configuration" formats arbitrarily

### ISSUE THREAD

**Issue by Zamiell** [streetsidesoftware/vscode-spell-checker#1940]
## Prettier

Lots of projects use [Prettier](https://prettier.io/) - including this one.

It is common for projects that use Prettier to validate in CI that all files are formatted correctly, like so:

```sh
npx prettier --check .
```

This command will check every file in the repository, and then throw and error if something isn't formatted correctly.

## The Problem With CSpell

When an end-user right-clicks on a misspelled word and selects "Add word to CSpell configuration", the CSpell VSCode extension will helpfully add it to the ignored words array, and then sort it in alphabetical order, and then format everything with 4 spaces.

What is not so helpful is that by default, Prettier uses 2 spaces, which means that CSpell doing this insertion will make the project break CI. This is a frustrating experience for developers and a hidden "gotcha", because when you edit a file with some trivial change, and you also happen to add a word to the dictionary at the same time, your "known-good" commit will fail CI.

To work around this, I could submit a PR that accomplishes the following:

1) After writing to a configuration/dictionary file, check if package.json exists.
2) If it exists, read it and check deps/devdeps for prettier.
3) If it exists, run prettier on the file path of the configuration/dictionary.

I think this is a better solution than fiddling with the CSpell formatting settings, because different projects could have different Prettier settings.

**Comment by Jason3S** [comment]
@Zamiell,

The spell checker tries to preserve formatting when editing a CSpell configuration file. It is possible that it is now breaking (I'll check). But the spell checker has no control over VS Code settings files.

Even so, the issue you talk about is easily solved with a commit hook. I think it is beyond the scope of the spell checker to read the configuration of other tools and try to emulate them.

**Comment by Zamiell** [comment]
> The spell checker tries to preserve formatting when editing a CSpell configuration file. 

It matches the amount of spaces but it doesn't match the array style. CSpell always put the array elements on the same line, whereas Prettier combines them on one line (when there are only a few elements).

**Comment by Jason3S** [comment]
> It matches the amount of spaces but it doesn't match the array style. CSpell always put the array elements on the same line, whereas Prettier combines them on one line (when there are only a few elements).

That is beyond the scope of the spell checker.

**Comment by Jason3S** [comment]
#2163 should fix this issue.
